include/zephyr/devicetree.h

```diff
@@ -1927,6 +1927,43 @@
 #define DT_FOREACH_CHILD(node_id, fn) \
 	DT_CAT(node_id, _FOREACH_CHILD)(fn)
 
+/**
+ * @brief Invokes "fn" for each child of "node_id" with a separator
+ *
+ * The macro "fn" must take one parameter, which will be the node
+ * identifier of a child node of "node_id".
+ *
+ * Example devicetree fragment:
+ *
+ *     n: node {
+ *             child-1 {
+ *                     label = "foo";
+ *             };
+ *             child-2 {
+ *                     label = "bar";
+ *             };
+ *     };
+ *
+ * Example usage:
+ *
+ *     const char *child_labels[] = {
+ *         DT_FOREACH_CHILD_SEP(DT_NODELABEL(n), DT_LABEL, (,))
```

Comment:
Let's not use DT_LABEL given the removal of those properties. How about DT_NODE_FULL_NAME instead?

Reply:
updated